cartoframes/dataset.py

```diff
@@ -105,6 +107,29 @@ def set_geodataframe(self, gdf):
     def get_table_name(self):
         return self._table_name
 
+    def get_dataset_info(self):
+        if self._is_saved_in_carto:
+            if self._dataset_info is None:
+                self._dataset_info = self._get_dataset_info()
+
+            return self._dataset_info
+        else:
+            raise CartoException('Your data is not synchronized with CARTO.'
+                                 'First of all, you should call upload method to save your data in CARTO.')
+
+    def set_dataset_info(self, dataset_info=None, privacy=None, name=None):
```

Comment:
Having the possibility to set a `dataset_info`, a `privacy` or a `name` in the same method looks a little bit weird to me.

The `DatasetInfo` wrapper looks good to me, maybe we can just remove `privacy` and `name` out of this method and force users to use `DatasetInfo`? 

Or if we want to go the `non-objected oriented` way, we can left only the `privacy` and `name` parameters.

Not sure what's best, thoughts?

Also, from the method name `set_dataset_info`, it does not look obvious to me that this is going to actually update the metadata.

Reply:
**- About dataset_info, a privacy or a name parameters:**
I did not find a better solution either, this is why we finally put all the cases. 

From the user point of view, I think the best option if the put only the properties to change:
```
dataset = Dataset.from_x
...
dataset.set_dataset_info(privacy=Dataset.PUBLIC)
```

But I would want to support also the case when the user is exploring the DatasetInfo (also it is more meaning accurated):
```
dataset = Dataset.from_x
...
dataset_info = dataset.get_dataset_info()
dataset_info.whatever
dataset.set_dataset_info(dataset_info)
```
Probably, we should go only with one of both...

**- About** 
 > it does not look obvious to me that this is going to actually update the metadata

I am hidding that part to the user on purpose. I think it is CARTO stuff and not relevant for the user.

Reply:
Maybe, for the future, is better to use **kwargs as arguments and raise warnings or even exceptions if the user is trying to update non-allowed properties.

Reply:
Yep, I like the idea about **kwargs if we go to the first one (properties without dataset_info).

Reply:
I'd check if API docs for a param like `**kwargs` (which may have a lot of information) integrate well with Jupyter notebooks when pressing Shift + Tab.

Reply:
And we would need to handle the properties in another place, so maybe we could move to the  `**kwargs` approach in the future if we need to expose a lot of properties.

From my side, it is more important the first question about dataset_info, properties or both

Reply:
One discussion we have had is if we should go with a more object-oriented design or a more functional one. Looks like in some cases we are using classes for better design, maintainability and for ease of use of developers and adding some helper methods for other kind of users more used to the functional style.

In this case we could have these two options:

For the case of people who want to work with classes:

```python
dataset = Dataset.from_x
...
dataset_info = dataset.get_dataset_info()
dataset_info.privacy = 'PUBLIC'
dataset_info.name = 'whatever'
dataset_info.save()
```

For the "functional" style:

```python
dataset = Dataset.from_x
dataset.update_dataset_info(privacy=..., name=...) 
```

Thoughts?